src/components/AppFrame/MenuItem.js

```diff
@@ -1,147 +1,127 @@
-export const AlertMessage = transition.div`
-	${alertStyling}
-	${transitionStatement}
+			"&::before": {
+				content: "",
+				position: "absolute",
+				top: "calc(10px + 0.2em)",
+				left: "-0.9em",
+				border: "solid transparent",
+				borderWidth: "0.4em 0.9em 0.4em 0",
+				borderRightColor: toastColor,
+			},
 
-	&:enter {
-		opacity: 0.01;
-		transform: translateX(-50%) scaleX(0);
-	}
-	&:enter-active {
-		opacity: 1;
-		transform: translateX(0) scaleX(1);
-	}
-	&:exit {
-		opacity: 1;
-		transform: translateX(0) scaleX(1);
-	}
-	&:exit-active {
-		opacity: 0.01;
-		transform: translateX(-50%) scaleX(0);
-	}
-`;
-AlertMessage.defaultProps = { timeout: 200, unmountOnExit: true, appear: true };
+			"&.show": {
+				opacity: 1,
+				transform: "translateX(0) scaleX(1)",
+			},
+		};
+	},
+	menuIcon: props => ({
+		fontSize: "24px",
+		verticalAlign: "middle",
+		color: props.active ? theme.palette.primary.light : theme.palette.text.hint,
+	}),
+	label: props => ({
+		fontFamily: theme.typography.button.fontFamily,
+		fontSize: "13px",
+		verticalAlign: "middle",
+		textTransform: "uppercase",
+		paddingLeft: "10px",
+		transition: "opacity 0.3s ease-out",
+		opacity: props.showLabel ? 1 : 0,
+	}),
+}));
 
-export const MenuIcon = styled(Icon)`
-	font-size: 24px;
-	vertical-align: middle;
-`;
+const FilteredLink = ({ staticContext, dispatch, component, ...props }) => <Link {...props} />;
 
-export const Label = styled.span`
-	font-family: ${getThemeProp(["fonts", "header"], "sans-serif")};
-	font-size: 13px;
-	vertical-align: middle;
-	text-transform: uppercase;
-	padding-left: 10px;
+const MenuItem = ({ open = false, label = "", icon, alert, isHidden = false, href, menuToggle, active, ...props }) => {
+	const classes = useStyles({ active: active, menuToggle: menuToggle, alertType: alert?.type, showLabel: open });
 
-	transition: opacity 0.3s ease-out;
-	opacity: ${ifFlag("show", 1, 0)};
-`;
-
-const MenuItem = ({ open = false, label = "", icon, alert, isHidden = false, href, ...props }) => {
-	let ItemWrapper = Block;
-	if (props.menuToggle) {
-		ItemWrapper = BlockWithA;
-	}
 	const alertMessage = useRef("");
+	const showAlert = !!alert;
+	const showAlertMessage = !!alert?.message;
 	if (alert && alert.message) {
 		alertMessage.current = alert.message;
 	}
 
+	if (isHidden) {
+		return null;
+	}
+
+	const ItemWrapper = menuToggle ? "a" : FilteredLink;
+
 	return (
-		!isHidden && (
-			<ItemWrapper to={href} {...props}>
-				<MenuIcon id={icon} />
-				{alert ? (
-					<Alert type={alert.type}>
-						<AlertMessage in={!!alert.message} type={alert.type}>
-							<Text message={alertMessage.current} />
-						</AlertMessage>
-					</Alert>
-				) : null}
-				<Label show={open}>
-					<Text message={label} />
-				</Label>
-			</ItemWrapper>
-		)
+		<ItemWrapper to={href} className={classes.block} {...props}>
+			<Icon id={icon} className={classes.menuIcon} />
+			<div className={`${classes.alert} ${showAlert ? "show" : undefined}`}>
```

Comment:
Why do you still render the <div/> section if "showAlert" is false, I definitely prefer the old syntax. I understand that without "show" class, if would not be visible, yet, the <div/> would still be renderer.

Only my opinion, but generally, I prefer the old syntax.